src/main/java/org/opensearch/knn/index/util/AbstractKNNLibrary.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.util;
+
+import org.opensearch.common.ValidationException;
+import org.opensearch.knn.index.KNNMethod;
+import org.opensearch.knn.index.KNNMethodContext;
+
+import java.util.Map;
+
+/**
+ * AbstractKNNLibrary implements common functionality shared between libraries
+ */
+public abstract class AbstractKNNLibrary implements KNNLibrary {
+
+    protected final Map<String, KNNMethod> methods;
+    protected final String version;
+
+    /**
+     * Constructor
+     *
+     * @param methods Map of k-NN methods that the library supports
+     * @param version String representing version of library
+     */
+    AbstractKNNLibrary(Map<String, KNNMethod> methods, String version) {
```

Comment:
lombok AllArgsConstructor?

Reply:
Thought of this but wanted to keep constructor package private. Didnt realize I could add access specification. Will update.